Unified build repository for F4MP.

blipvert opened the pull request:
Pull request to implement issue #4

blipvert committed:
Initial unifed CMake project for server, client and launcher.

blipvert committed:
Require CMake 3.8 for C# project support.

blipvert committed:
Removed launcher submodule as it is not yet ready.

blipvert committed:
Removed legacy code, now on separate branch.

blipvert committed:
Removed most .gitignore patterns since they don't have effect in submodules.

blipvert committed:
Added .gitignore patterns for VS Code and default CMake build directory.

blipvert committed:
Added project version number and description.

blipvert committed:
Updated information in README.md.

blipvert committed:
Minor edit to README.

0rdinal reviewed (approved):
Good work submitting an issue and tracking on the project. Approved.

blipvert committed:
Target `develop` branch of client and server.

blipvert committed:
Commited submodule changes:

* client d981f83...c2a3dcd (1):
  > Changed CMAKE_SOURCE_DIR to CMAKE_CURRENT_SOURCE_DIR. (blipvert)

* server fbbce15...b3ce770 (1):
  > Changed CMAKE_SOURCE_DIR to CMAKE_CURRENT_SOURCE_DIR. (blipvert)

benkyd reviewed (approved):
Looks good
